### PR TITLE
IGVF-1241-update-analysis-set

### PR DESCRIPTION
**Pull request by ian-whaling**

### CODE REVIEW

File: src/igvfd/schemas/analysis_set.json [hunk 1]
```diff
@@ -38,28 +38,56 @@
     "dependentSchemas": {
         "$merge": [
             "file_set.json#/dependentSchemas"
-        ]
+        ],
+        "file_set_type": {
```
Comment (gabdank) [inline src/igvfd/schemas/analysis_set.json:42]: I think you need to add file_set_type to required properties

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/tests/test_upgrade_analysis_set.py [hunk 2]
```diff
@@ -13,3 +13,9 @@ def test_analysis_set_upgrade_1_2(upgrader, analysis_set_v1):
     assert 'input_file_set' not in value
     assert input_file_sets == value['input_file_sets']
     assert value['schema_version'] == '2'
+
+
+def test_analysis_set_upgrade_3_4(upgrader, analysis_set_v3):
+    value = upgrader.upgrade('analysis_set', analysis_set_v3, current_version='3', target_version='4')
```
Comment (gabdank) [inline src/igvfd/tests/test_upgrade_analysis_set.py:19]: please add validation of preexisting file_set_type

File: src/igvfd/schemas/analysis_set.json (hunk 1 above)
Comment (ian-whaling) [reply src/igvfd/schemas/analysis_set.json:42]: fixed

File: src/igvfd/upgrade/file_set.py [hunk 3]
```diff
@@ -121,3 +121,12 @@ def measurement_set_8_9(value, system):
     if 'sequencing_library_type' in value:
         value['sequencing_library_types'] = value['sequencing_library_type']
         del value['sequencing_library_type']
+
+
+@upgrade_step('analysis_set', '3', '4')
+def analysis_set_3_4(value, system):
+    # https://igvf.atlassian.net/browse/IGVF-1241
+    value['file_set_type'] = 'intermediate analysis'
```
[the remark this answers is not in the document]
Comment (ian-whaling) [reply src/igvfd/upgrade/file_set.py]: added

File: src/igvfd/tests/test_upgrade_analysis_set.py (hunk 2 above)
Comment (ian-whaling) [reply src/igvfd/tests/test_upgrade_analysis_set.py:20]: added

**Review by gabdank — APPROVED** [review]

**Pull request merged**